flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -628,12 +628,19 @@ public boolean dropTemporaryView(String path) {
 
     @Override
     public String[] listUserDefinedFunctions() {
-        return functionCatalog.getUserDefinedFunctions();
+        return sortFunctions(functionCatalog.getUserDefinedFunctions());
     }
 
     @Override
     public String[] listFunctions() {
-        return functionCatalog.getFunctions();
+        return sortFunctions(functionCatalog.getFunctions());
```

Comment:
I think we can simply `Arrays.sort(functionCatalog.getFunctions())`. The returned `getFunctions` should never be null.